cogip/widgets/wizard.py

```diff
@@ -0,0 +1,353 @@
+        send_button.clicked.connect(self.send)
+
+    @qtSlot()
+    def send(self, clicked: bool):
+        """
+        Send chosen value to parent dialog on Send button click.
+        """
+        self.response.emit(self.buttons.checkedButton().text())
+
+
+class SelectWizard(QtCore.QObject):
+    """
+    SelectWizard class.
+
+    Build a widget to select one or more integer, float or string from a list.
+    """
+    response: qtSignal = qtSignal(list)
+
+    def __init__(self, wizard: dict[str, Any], parent: QtWidgets.QWidget):
+        """
+        Class constructor.
+
+        Arguments:
+            wizard: wizard request
+            parent: the parent widget
+        """
+        super().__init__()
+        self.wizard = wizard
+
+        layout = QtWidgets.QVBoxLayout()
+        parent.setLayout(layout)
+
+        self.buttons = []
+        for v in wizard["choices"]:
+            button = QtWidgets.QCheckBox(str(v))
+            self.buttons.append(button)
+            button.setChecked(v in wizard["value"])
+            layout.addWidget(button)
+        send_button = QtWidgets.QPushButton("Send")
+        layout.addWidget(send_button)
+        send_button.clicked.connect(self.send)
+
+    @qtSlot()
+    def send(self, clicked: bool):
+        """
+        Send chosen value to parent dialog on Send button click.
+        """
+        print([button.text() for button in self.buttons if button.isChecked()])
+        self.response.emit([button.text() for button in self.buttons if button.isChecked()])
+
+
+class CampWizard(QtCore.QObject):
+    """
+    CampWizard class.
+
+    Build a widget to select a string.
+    """
+    response: qtSignal = qtSignal(str)
+
+    def __init__(self, wizard: dict[str, Any], parent: QtWidgets.QWidget):
+        """
+        Class constructor.
+
+        Arguments:
+            wizard: wizard request
+            parent: the parent widget
+        """
+        super().__init__()
+        self.wizard = wizard
+
+        layout = QtWidgets.QVBoxLayout()
+        parent.setLayout(layout)
+
+        button_layout = QtWidgets.QHBoxLayout()
+        layout.addLayout(button_layout)
+
+        self.buttons = QtWidgets.QButtonGroup()
+
+        self.button_blue = QtWidgets.QRadioButton()
+        self.button_blue.setObjectName("blueCamp")
+        self.button_blue.setCheckable(True)
+        self.button_blue.setStyleSheet(
+            """
+            QRadioButton#blueCamp {
+                border-color: #005CE6;
+                background-color: #005CE6;
+                border-width: 2px;
+                border-radius: 10px;
+                border-style: inset;
+                min-width: 6em;
+                padding: 6px;
+            }
+            QRadioButton#blueCamp:checked {
+                border-color: beige;
+                border-style: outset;
+            }
+            QRadioButton#blueCamp::indicator {
+                border-width: 0;
+            }
+            """)
```

Comment:
Just a question, not more simple to put this in a seperate file ?

Reply:
No that's the way to do it in Qt...